django_irods/storage.py

```diff
@@ -136,334 +90,214 @@ def unzip(self, zip_file_path, unzipped_folder=""):
-            )
-        # remove potential '\n' from stdout
-        size_string = stdout.replace("\n", "")
-        try:
-            ret = int(float(size_string))
-            return ret
-        except ValueError:
-            return 0
+        bucket, name = bucket_and_name(s3_bucket_name)
+        s3_object = self.connection.Object(bucket, name)
+        return s3_object.e_tag.strip('"')
 
-    def checksum(self, full_name, force_compute=True):
+    def download_file(self, s3_bucket_name, local_file_path):
         """
-        Compute/Update checksum of file object and return the checksum
-        :param full_name: the data object name with full collection path in order to locate data object from current
-        working directory
-        :return: checksum of the file object
+        Download file from S3 bucket to local file path
+        :param s3_bucket_name: the data object name with full collection path in order to locate data object from
+        current working directory
+        :param local_file_path: the local file path to download the file to
         """
-        # first force checksum (re)computation
-        if force_compute:
-            self.session.run("ichksum", None, "-f", full_name)
-        # retrieve checksum using iquest
-        # get data object name only from the full_name input parameter to be used by iquest
-        if "/" in full_name:
-            file_info = full_name.rsplit("/", 1)
-            coll_name_query = IrodsStorage.get_absolute_path_query(file_info[0])
-            obj_name = file_info[1]
-        else:
-            coll_name_query = "COLL_NAME = '{}'".format(settings.IRODS_HOME_COLLECTION)
-            obj_name = full_name
-
-        qrystr = "SELECT DATA_CHECKSUM WHERE {} AND DATA_NAME = '{}'".format(
-            coll_name_query, obj_name
-        )
-        stdout = self.session.run("iquest", None, "%s", qrystr)[0]
-        if "CAT_NO_ROWS_FOUND" in stdout:
-            raise ValidationError(
-                "{} cannot be found in iRODS to retrieve " "checksum".format(obj_name)
-            )
-        # remove potential '\n' from stdout
-        checksum = stdout.strip("\n")
-        if not checksum:
-            if force_compute:
-                raise ValidationError(
-                    "checksum for {} cannot be found in iRODS".format(obj_name)
-                )
-            # checksum hasn't been computed, so force the checksum computation
-            return self.checksum(full_name, force_compute=True)
-        return checksum
+        bucket, name = bucket_and_name(s3_bucket_name)
+        self.connection.Bucket(bucket).download_file(name, local_file_path)
+
+    def signed_url(self, name):
+        super_url = super().url(name.strip("/"))
+        # if super_url.startswith("http://minio:9000"):  # TODO make this based on DEBUG setting?
+        #    return super_url.replace("http://minio:9000", "http://localhost:9000")
```

Comment:
Uncomment these lines for downloads on a lcaol run. I keep it as minio for the unit tests. This will need to be refined into configuration at some point.

Reply:
I think it makes sense to use DEBUG setting for this.
Or add a separate setting like `STORAGE_HOST` that defaults to localhost